### ISSUE TITLE
`prefer-set-has`: Add option for number of elements required

### ISSUE THREAD

**Issue by RebeccaStevens** [sindresorhus/eslint-plugin-unicorn#2473]
### Description

From the docs:

> [Set#has()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set/has) is faster than [Array#includes()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/includes).

This is only true for sets containing a larger number of elements. For a small number of elements, using `array.includes` is both faster for look-ups, and has a smaller memory footprint.

I would like to propose adding an option to this rule to specify how many elements are needed before this rule kicks in. Additionally, it would be nice if the rule enforced using an array over a set when this threshold isn't met.

I suggest the name `threshold` for this option.
If this option were to be enabled by default, then the default should probably be somewhere between 10 and 30.

### Fail

```js
const array = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15];
const hasValue = value => array.includes(value);
```

```js
const set = new Set([1, 2, 3]);
const hasValue = value => set.has(value);
```

### Pass

```js
const array = [1, 2, 3];
const hasValue = value => array.includes(value);
```

```js
const set = new Set([1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15]);
const hasValue = value => set.has(value);
```

**Comment by sindresorhus** [comment]
> This is only true for sets containing a larger number of elements. For a small number of elements, using array.includes is both faster for look-ups, and has a smaller memory footprint.

Source?
